### PR TITLE
vm: support V3 and static syscalls in VmInterpHarness

### PR DESCRIPTION
**Pull request by 0x0ece**

### CODE REVIEW

File: src/vm_interp.rs [hunk 1]
```diff
@@ -117,44 +117,62 @@ pub fn execute_vm_interp(syscall_context: SyscallContext) -> Option<SyscallEffec
...
-    let mut stubbed_syscall_reg = FunctionRegistry::<BuiltinFunction<TestContextObject>>::default();
-
-    for (key, (name, _)) in syscall_reg.iter() {
-        stubbed_syscall_reg
-            .register_function(key, name, SyscallStub::vm)
-            .unwrap();
-    }
-
+    let enable_stack_frame_gaps = (sbpf_version == SBPFVersion::V0)
+        || !feature_set.is_active(&bpf_account_data_direct_mapping::id());
```
Comment (topointon-jump) [inline src/vm_interp.rs]: Doesn't this mean that we will never test SBPF V0 programs with stack frame gaps disabled?

When direct mapping is activated everywhere (soon), on real networks most programs will be SBPF V0, and will be executed with stack frame gaps disabled.

Agave uses `enable_stack_frame_gaps: !feature_set.is_active(&bpf_account_data_direct_mapping::id())` - what's the reason for the different condition here?

File: src/vm_interp.rs [hunk 2]
```diff
@@ -317,16 +317,21 @@ non-issue here, but require explicit handling in FD. This causes a slight
...
     let mut fn_reg = FunctionRegistry::default();
     let max_pc = vm_ctx.rodata.len() / 8;
 
     // register entry point
-    let _ = fn_reg.register_function(
-        ebpf::hash_symbol_name(b"entrypoint"),
-        b"entrypoint",
-        vm_ctx.entry_pc as usize,
-    );
+    let entry_pc = (vm_ctx.entry_pc as usize).min(max_pc.saturating_sub(1));
```
Comment (topointon-jump) [inline src/vm_interp.rs:330]: What's the reasoning behind bounding the entry PC here? Is this to mimic Agave's behaviour?

File: src/vm_interp.rs (hunk 1 above)
Comment (0x0ece) [reply src/vm_interp.rs]: in V1+ gaps must be disabled (the stack is reversed).
We can test V0 with and without gaps, but starting from V1 gaps must be disabled.
Agave is prob assuming that direct mapping will be enabled before V1?

File: src/vm_interp.rs (hunk 2 above)
Comment (0x0ece) [reply src/vm_interp.rs:330]: An invalid entry_pc is discarded by elf parsing. If we leave an invalid entry_pc here, there are situations where solfuzz-agave (the jit) crashes.

File: src/vm_interp.rs (hunk 1 above)
Comment (0x0ece) [reply src/vm_interp.rs]: update: the expression was incorrect, it should be `&&` not `||`.
gaps are disabled in v>v0 OR dm enabled
so gaps are enabled if v==v0 && dm is disabled

**Review by mjain-jump — APPROVED** [review]

**Pull request merged**